pkg/codegen/nodejs/gen.go

```diff
@@ -1417,6 +1415,81 @@ func GeneratePackage(tool string, pkg *schema.Package, extraFiles map[string][]b
 		types.typeDetails, types.types = typeDetails, typeList
 	}
 
+	return modules, nil
+}
+
+// LanguageResource holds information about a resource to be used by downstream codegen.
+type LanguageResource struct {
+	*schema.Resource
+
+	Name       string             // The resource name (e.g., "FlowSchema")
+	Package    string             // The name of the package containing the resource definition (e.g., "flowcontrol.v1alpha1")
+	Properties []LanguageProperty // Properties of the resource
+}
+
+// LanguageProperty holds information about a resource property to be used by downstream codegen.
+type LanguageProperty struct {
+	ConstValue string // If set, the constant value of the property (e.g., "flowcontrol.apiserver.k8s.io/v1alpha1")
+	Name       string // The name of the property (e.g., "FlowSchemaSpec")
+	Package    string // The package path containing the property definition (e.g., "outputs.flowcontrol.v1alpha1")
+}
+
+// LanguageResources returns a map of resources that can be used by downstream codegen. The map
+// key is the resource schema token.
+func LanguageResources(tool string, pkg *schema.Package) (map[string]LanguageResource, error) {
```

Comment:
Does the `tool` parameter doesn't influence the output? If not, can we remove it (and just pass `""` to `generateModuleContextMap`?

```suggestion
func LanguageResources(pkg *schema.Package) (map[string]LanguageResource, error) {
```

Reply:
Nope, looks like we can remove it.